flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorTest.java

```diff
@@ -1845,6 +1860,42 @@ public void testSlotReportDoesNotContainStaleInformation() throws Exception {
 		}
 	}
 
+	@Test
+	public void testSlotReportResourceProfile() throws Throwable {
```

Comment:
I think this test shows that the abstractions are too intermingled.

The code that was added in TaskExecutor was a change to replace the UNKNOWN profile attached to a slot by a profile with proper values. The code only takes the task manager's config (or managed memory size), and determines the size of the slots.

Such a change should not need the creation of a full TaskExecutor to test it. That means something is not right in the abstraction.

This test setup means that every time the TaskExecutor setup changes, the test for the simple code "config -> profile" needs to be adjusted, which contributes to making maintenance complex.